pkg/process/process.go

```diff
@@ -0,0 +1,119 @@
+package process
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/shirou/gopsutil/v4/process"
+)
+
+type Manager interface {
+	ReadPidFile() (int, error)
+	Name() string
+	PidFilePath() string
+	Exists() bool
+	Terminate() error
+	Kill() error
+	FindProcess() (*os.Process, error)
+	WritePidFile(pid int) error
+}
+
+type Process struct {
+	name           string
+	pidFilePath    string
+	executablePath string
+}
+
+func New(name, pidFilePath, executablePath string) (*Process, error) {
+	return &Process{name: name, pidFilePath: pidFilePath, executablePath: executablePath}, nil
+}
+
+func (p *Process) Name() string {
+	return p.name
+}
+
+func (p *Process) PidFilePath() string {
+	return p.pidFilePath
+}
+
+func (p *Process) ExecutablePath() string {
+	return p.executablePath
+}
+
+func (p *Process) ReadPidFile() (int, error) {
+	data, err := os.ReadFile(p.PidFilePath())
+	if err != nil {
+		return -1, err
+	}
+	pidStr := strings.TrimSpace(string(data))
+	pid, err := strconv.Atoi(pidStr)
+	if err != nil {
+		return -1, fmt.Errorf("invalid pid file: %v", err)
+	}
+	return pid, nil
+}
+
+func (p *Process) FindProcess() (*process.Process, error) {
```

Comment:
Does it need to be public or it an be a private helper? Exposing the underlying process package in this packages means we can never change the implementation, for example using another package to validate the process executable path.

Reply:
In crc, there's a codepath that uses something like `FindProcess` to get the underlying Process object:
https://github.com/crc-org/crc/compare/main...vyasgun:crc:test-process-lib

I think this can be changed in crc so the code doesn't use the underlying `process.Process` object if it's not recommended to return it.